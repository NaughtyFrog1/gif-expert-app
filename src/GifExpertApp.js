import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'

const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    'One Punch',
    'Samurai X',
    'Dragon Ball',
  ])

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <li key={category}>{category}</li>
        ))}
      </ol>
    </>
  )
}

export default GifExpertApp

/**
 * Cada elemento en una lista debe tener una "key" prop única.
 * Se recomienda que esta "key" no sea el índice del elemento, ya que si se modifica el array pueden producirse errores
 */
