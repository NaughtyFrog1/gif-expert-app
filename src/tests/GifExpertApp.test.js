import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('test for GifExpertApp', () => {
  test('should render the component properly', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should display a list of categories', () => {
    const categories = ['The Office', 'Brooklyn nine-nine']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
