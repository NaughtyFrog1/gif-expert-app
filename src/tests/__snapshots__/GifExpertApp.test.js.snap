// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for GifExpertApp should display a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <section>
    <GifGrid
      category="The Office"
      key="The Office"
    />
    <GifGrid
      category="Brooklyn nine-nine"
      key="Brooklyn nine-nine"
    />
  </section>
</Fragment>
`;

exports[`test for GifExpertApp should render the component properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <section />
</Fragment>
`;
